dspace-api/src/main/java/org/dspace/app/canvasdimension/CanvasDimensionCLI.java

```diff
@@ -0,0 +1,203 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.canvasdimension;
+
+import java.util.Arrays;
+import java.util.UUID;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.MissingArgumentException;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.dspace.app.canvasdimension.factory.IIIFCanvasDimensionServiceFactory;
+import org.dspace.app.canvasdimension.service.IIIFCanvasDimensionService;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.DSpaceObject;
+import org.dspace.content.Item;
+import org.dspace.core.Constants;
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.eperson.factory.EPersonServiceFactory;
+import org.dspace.eperson.service.EPersonService;
+import org.dspace.handle.factory.HandleServiceFactory;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+
+/**
+ * Sets IIIF canvas metadata on bitstreams based on image size.
+ *
+ * @author Michael Spalti mspalti@willamette.edu
+ */
+public class CanvasDimensionCLI {
+
+    private static final EPersonService epersonService = EPersonServiceFactory.getInstance().getEPersonService();
+    private static final ConfigurationService configurationService = DSpaceServicesFactory.getInstance()
+                                                                                          .getConfigurationService();
+
+    private CanvasDimensionCLI() {}
+
+    public static void main(String[] argv) throws Exception {
+
+
+        boolean iiifEnabled = configurationService.getBooleanProperty("iiif.enabled");
+        if (!iiifEnabled) {
+            System.out.println("IIIF is not enabled on this DSpace server.");
+            System.exit(0);
+        }
```

Comment:
this is not strictly needed. The script has a value also when iiif is not enabled because it adds technical metadata at the bitstream level. It is true that without IIIF the API service will be unable to return any information but this is just the fallback for the scenario where the ImageIO class is not able to calculate it.
I would suggest to just print a warning message instead than refuse to run the script